### ISSUE TITLE
Filename with .js extension fails in webpack@5

### ISSUE THREAD

**Issue by KarunaLakshman** [FormidableLabs/webpack-stats-plugin#91]
**Versions:**

- webpack@5.74.0
- webpack-stats-plugin@1.1.0

**Problem:**

To meet the project requirement, I'm supposed to generate `stats.js` file instead of the default `stats.json` file.  Unfortunately the below setup ended up with an error.

```
new StatsWriterPlugin({
  filename: 'stats.js',
  fields: ['entrypoints'],
  transform(data) {
    const assets = data.entrypoints.main.assets;
    const stats = assets.map((asset, index) => {
      const order = index + 1;
      return {
        id: `asset-${order}`,
        name: asset.name,
        order
      };
    });
    statsStr = JSON.stringify(stats);
    return `define('stats',function(){return ${statsStr}});`;
  }
})
```
**Error:**

```
[webpack-cli] HookWebpackError: inputSource.sourceAndMap is not a function
    at makeWebpackError (/Users/xuser/workspace/my-project/node_modules/webpack/lib/HookWebpackError.js:48:9)
    at /Users/xuser/workspace/my-project/node_modules/webpack/lib/Compilation.js:3060:12
    at eval (eval at create (/Users/xuser/workspace/my-project/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:20:1)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
-- inner error --
TypeError: inputSource.sourceAndMap is not a function
    at /Users/xuser/workspace/my-project/node_modules/terser-webpack-plugin/dist/index.js:454:27
    at next (/Users/xuser/workspace/my-project/node_modules/terser-webpack-plugin/dist/utils.js:81:7)
    at Array.forEach (<anonymous>)
    at /Users/xuser/workspace/my-project/node_modules/terser-webpack-plugin/dist/utils.js:84:26
    at new Promise (<anonymous>)
    at throttleAll (/Users/xuser/workspace/my-project/node_modules/terser-webpack-plugin/dist/utils.js:53:10)
    at TerserPlugin.optimize (/Users/xuser/workspace/my-project/node_modules/terser-webpack-plugin/dist/index.js:662:11)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
caused by plugins in Compilation.hooks.processAssets
TypeError: inputSource.sourceAndMap is not a function
    at /Users/xuser/workspace/my-project/node_modules/terser-webpack-plugin/dist/index.js:454:27
    at next (/Users/xuser/workspace/my-project/node_modules/terser-webpack-plugin/dist/utils.js:81:7)
    at Array.forEach (<anonymous>)
    at /Users/xuser/workspace/my-project/node_modules/terser-webpack-plugin/dist/utils.js:84:26
    at new Promise (<anonymous>)
    at throttleAll (/Users/xuser/workspace/my-project/node_modules/terser-webpack-plugin/dist/utils.js:53:10)
    at TerserPlugin.optimize (/Users/xuser/workspace/my-project/node_modules/terser-webpack-plugin/dist/index.js:662:11)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
```

**Tips:** 

I found  `curCompiler.webpack` at [line 157](https://github.com/FormidableLabs/webpack-stats-plugin/blob/main/lib/stats-writer-plugin.js#L157 ) goes undefined, so I believe [source](https://github.com/FormidableLabs/webpack-stats-plugin/blob/main/lib/stats-writer-plugin.js#L161-L168) object needs additional property (sourceAndMap).

**Comment by ryan-roemer** [comment]
`curCompiler.webpack` is part of a ternary check that allows it to be false-y. Also, `sourceAndMap` isn't used by this plugin at all. From your stack trace it looks like you have an error in the `terser-webpack-plugin` and not this plugin.

If you diagnose further and think that this plugin is somehow involved, can you open a minimal repository with install, build / error reproduction steps? Thanks!

**Comment by ryan-roemer** [comment]
Ah -- so it may be the case that `stats.js` is then getting minified. What I'd suggest is taking this plugin out of your config first. Manually drop in the created JS file you're expecting to create and then verify it can be minified correctly by terser.

**Comment by ryan-roemer** [comment]
OK -- I have reproduced this one. Weird.

Anyways, issue is verified and I'll look in further. Thanks!

**Comment by ryan-roemer** [comment]
@KarunaLakshman if you have a moment, can you pull down and try out https://github.com/FormidableLabs/webpack-stats-plugin/pull/92 ? Thanks!

**Comment by KarunaLakshman** [comment]
@ryan-roeme, I really appreciate your quick act and thank you! Sure I'll try it today and let you know

**Comment by KarunaLakshman** [comment]
Hey @ryan-roemer, I tested your changes and it worked great. Thanks once again for the fix and the cool plugin.

**Comment by ryan-roemer** [comment]
Fix released in `webpack-stats-plugin@1.1.1`
